app/controllers/recipes_controller.rb

```diff
@@ -63,7 +63,24 @@ def set_recipe
 
   # Only allow a list of trusted parameters through.
   def recipe_params
-    params.require(:recipe).permit(:name, :ingredients, :directions, :yield, :prep_time, :cook_time, :description,
-                                   :rating, :is_favorite, :notes, :category_ids, :categories)
+    params.require(:recipe).permit(recipe_fields).tap do |p|
+      p['category_names'] = p.fetch('category_names', '').split(',').map(&:strip)
```

Comment:
Converts `category_params` from a string of comma-separated category names to an array of category name strings.